src/lib.rs

```diff
@@ -11,6 +11,678 @@
+                    mqtt::PropertyCode::UserProperty,
+                    COMMSTATUS_NUM,
+                    &commstatus_val.to_string(),
+                )
+                .map_err(|_| {
+                    UStatus::fail_with_code(UCode::INTERNAL, "Unable to add comm_status property")
+                })?;
+        }
+
+        // Map UAttributes.reqid
+        // TODO: Need to make this non-optional
+        if let Some(reqid) = attributes.reqid.as_ref() {
+            let reqid_val = reqid.to_hyphenated_string();
+            properties
+                .push_string_pair(mqtt::PropertyCode::UserProperty, REQID_NUM, &reqid_val)
+                .map_err(|_| {
+                    UStatus::fail_with_code(UCode::INTERNAL, "Unable to add reqid property")
+                })?;
+        }
+
+        // Map Optional UAttributes.token
+        if let Some(token_val) = attributes.token.as_ref() {
+            properties
+                .push_string_pair(mqtt::PropertyCode::UserProperty, TOKEN_NUM, token_val)
+                .map_err(|_| {
+                    UStatus::fail_with_code(UCode::INTERNAL, "Unable to add token property")
+                })?;
+        }
+
+        // Map Optional UAttributes.traceparent
+        if let Some(traceparent_val) = attributes.traceparent.as_ref() {
+            properties
+                .push_string_pair(
+                    mqtt::PropertyCode::UserProperty,
+                    TRACEPARENT_NUM,
+                    traceparent_val,
+                )
+                .map_err(|_| {
+                    UStatus::fail_with_code(UCode::INTERNAL, "Unable to add traceparent property")
+                })?;
+        }
+
+        // Map UAttributes.payload_format
+        let payload_fmt_val = attributes.payload_format.value();
+        properties
+            .push_string_pair(
+                mqtt::PropertyCode::UserProperty,
+                PAYLOAD_NUM,
+                &payload_fmt_val.to_string(),
+            )
+            .map_err(|_| {
+                UStatus::fail_with_code(UCode::INTERNAL, "Unable to add payload_format property")
+            })?;
+
+        Ok(properties)
+    }
+
+    /// Get UAttributes from mqtt header properties.
+    ///
+    /// # Arguments
+    /// * `props` - Mqtt properties to get UAttributes from.
+    fn get_uattributes_from_mqtt_properties(
+        props: &mqtt::Properties,
+    ) -> Result<UAttributes, UStatus> {
+        let mut attributes = UAttributes::default();
+
+        props.user_iter().for_each(|(key, value)| {
+            match key.as_str() {
+                ID_NUM => {
+                    let id = UUID::from_str(&value)
+                        .map_err(|_| {
+                            UStatus::fail_with_code(
+                                UCode::INTERNAL,
+                                "Unable to parse id from mqtt properties",
+                            )
+                        })
+                        .unwrap();
```

Comment:
If we .unwrap() on the err path would this cause a run-time panic?

Reply:
Resolved these, there was a bit of a rewrite but ensured any unwraps were managed appropriately.